Mathlib/Algebra/Lie/DirectSum.lean

```diff
@@ -0,0 +1,252 @@
+/-
+Copyright (c) 2020 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+
+! This file was ported from Lean 3 source module algebra.lie.direct_sum
+! leanprover-community/mathlib commit c0cc689babd41c0e9d5f02429211ffbe2403472a
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.DirectSum.Module
+import Mathlib.Algebra.Lie.OfAssociative
+import Mathlib.Algebra.Lie.Submodule
+import Mathlib.Algebra.Lie.Basic
+
+/-!
+# Direct sums of Lie algebras and Lie modules
+
+Direct sums of Lie algebras and Lie modules carry natural algebra and module structures.
+
+## Tags
+
+lie algebra, lie module, direct sum
+-/
+
+
+universe u v w w₁
+
+namespace DirectSum
+
+open Dfinsupp
+
+open scoped DirectSum
+
+variable {R : Type u} {ι : Type v} [CommRing R]
+
+section Modules
+
+/-! The direct sum of Lie modules over a fixed Lie algebra carries a natural Lie module
+structure. -/
+
+
+variable {L : Type w₁} {M : ι → Type w}
+
+variable [LieRing L] [LieAlgebra R L]
+
+variable [∀ i, AddCommGroup (M i)] [∀ i, Module R (M i)]
+
+variable [∀ i, LieRingModule L (M i)] [∀ i, LieModule R L (M i)]
+
+instance : LieRingModule L (⨁ i, M i) where
+  bracket x m := m.mapRange (fun i m' => ⁅x, m'⁆) fun i => lie_zero x
+  add_lie x y m := by
+    refine' Dfinsupp.ext fun _ => _ -- Porting note: Originally `ext`
```

Comment:
```suggestion
    refine' Dfinsupp.ext fun _ => _ -- Porting note: Originally `ext`
```
You might be able to fix this by changing the priority of the `ext` lemma `Dfinsupp.ext`.

Reply:
Actually, I take that back. The problem is that you need to add an `ext` lemma for `DirectSum` because the new `ext` tactic won't unfold as much and won't apply `Dfinsupp.ext`.